Addressed issue #36 by using ShellCheck

thereadonly opened the pull request:
Committer: sdk @thereadonly 

### Summary
What does this pull request do? 
Tidy up the script by clearing up the unquoted paths that could cause issues.

### Implementation details
How are the changes [implemented?]
The changes are implemented using ShellCheck - https://www.shellcheck.net to tidy up the script. Further changes to this script should be run through shellcheck first and then merged.

Requesting review on the fixes and the existing suggestions by ShellCheck.

#### Issues 
Before fix : [https://gist.github.com/thereadonly/9d02c09a1b621886b2a4e373f42e56af](url)
After fix : [https://gist.github.com/thereadonly/f736ca2bc9f74280ef94c9e2cf32cb11](url)

###  Testing
TBD post review suggestion on above issues.

### Licensing
This contribution is under the terms of the Apache 2.0 License:  yes

Commit:
Addressed issue #36 by using ShellCheck

 Committer: sdk <dinesh@thereadonly.com>

jahkeup reviewed (approved):
Thanks for tidying this up @thereadonly! We'll be enabling shellcheck as a result of the effort here :+1:

jahkeup commented on ecs-logs-collector.sh:
```diff
@@ -213,11 +213,11 @@ pack() {
 
-  cd ${curdir}
-  ${tar_bin} -czf ${infodir}.tgz ${infodir} > /dev/null 2>&1
+  cd "$curdir" || echo "cd failed."; exit 
```
This will need some braces around it to prevent the unconditional exit:
```suggestion
  cd "$curdir" || { echo "cd failed."; exit 1; }
```

jahkeup commented on ecs-logs-collector.sh:
```diff
@@ -172,7 +172,7 @@ try_set_instance_infodir() {
       infodir="${infodir}/${instance_id}"
       info_system="${infodir}/system"
-      echo "$instance_id" | $info_system/instance-id.txt
+      echo "$instance_id" | "$info_system"/instance-id.txt
```
Looks like this was missed in the original review of this section:
```suggestion
      mkdir -p "$info_system"
      echo "$instance_id" > "$info_system"/instance-id.txt
```

jahkeup commented on ecs-logs-collector.sh:
```diff
@@ -213,11 +213,11 @@ pack() {
 
   local tar_bin
-  tar_bin="`which tar 2>/dev/null`"
+  tar_bin="$(which tar 2>/dev/null)"
```
Let's combine this line with the above:

```bash
local tar_bin="$(which tar 2>/dev/null)"
```

jahkeup replied:
actually, shellcheck likes these on two lines. So okay.

jahkeup reviewed (changes requested):
Taking a closer look, there's a few small, but necessary, changes that we'll want to make before merging this.

jahkeup committed:
Use builtins and update handling errors from cd

jahkeup committed:
Handle command line arguments as elements

jahkeup committed:
Fix handling of instance id in collection tarball

jahkeup committed:
Fix intention globbing fixed by shellcheck

This was taking advantage of the glob expansion inside the loop, not
in the iterator. This is changed to expand in the iterator prior to
use, which for now is OK because there are not any files with spaces
in them.

jahkeup committed:
Fix shellcheck errors from rebase

jahkeup reviewed (approved):
Thanks for your contribution @thereadonly! :fireworks:
